server/events/models/models.go

```diff
@@ -131,6 +131,11 @@ func NewRepo(vcsHostType VCSHostType, repoFullName string, cloneURL string, vcsU
 		return Repo{}, fmt.Errorf("invalid repo format %q, repo %q should not contain any /'s", repoFullName, owner)
 	}
 
+	azuredevopsVcsToken := ""
+	if vcsHostType == AzureDevops {
```

Comment:
Could we allow passing in a custom token via flag and then remove this if block?

Reply:
So I think we need a flag to indicate that Azure DevOps is on-premise - this can be derived when AzureDevopsHostName != "dev.azure.com" or could be manually passed? If derived at what level do you think we should do this?